conn_test.go

```diff
@@ -195,14 +212,152 @@ func TestConnectionPool_Broadcast(t *testing.T) {
 		},
 	}
 
-	cp.Broadcast(msg)
+	cp.BroadcastSingleMessage(msg)
+
+	// wait until receive the msg
+	wg.Wait()
+}
+
+func Execute_BroadcastMultipleMessage(textList []string, waitSize int) ([]bool, error) {
+	cp := cleisthenes.NewConnectionPool()
+
+	cAddressList := make([]cleisthenes.Address, 3)
+
+	for i, _ := range cAddressList {
+		cAddressList[i] = cleisthenes.Address{"127.0.0.1", GetAvailablePort(8000)}
+	}
+
+	connList := make([]cleisthenes.Connection, 3)
+	var err error
+	err = nil
+
+	for i, _ := range connList {
+		connList[i], err = cleisthenes.NewConnection(cAddressList[i], "TestConnection"+strconv.Itoa(i+1), mock.NewStreamWrapper())
+		if err != nil {
+			// return error when error occurred at making connection
+			return nil, err
+		}
+	}
+
+	for _, conn := range connList {
+		cp.Add(conn.Id(), conn)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(waitSize)
+
+	checkReceive := make([]bool, len(textList))
```

Comment:
msgReceivedChecker, msgReceivedCheckList? anyway It would be better variable name ends with *noun*